File: ulohy/14-baleni-do-krabice/reseni/main.py
if __name__ == "__main__":
    box_volume = float(input())
    input()

    total_objects_volume = 0
    current_object_volume = 0

    while total_objects_volume + current_object_volume <= box_volume and current_object_volume != -1:
        total_objects_volume += current_object_volume
        current_object_volume = float(input())

    print(f"{total_objects_volume:g}")

File: ulohy/14-baleni-do-krabice/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let boxVolume;
let totalObjectsVolume = 0;

rl.on("line", line => {
    if (boxVolume === undefined) {
        boxVolume = parseFloat(line);
        return;
    }

    if (line === "") {
        return;
    }

    const currentObjectVolume = parseFloat(line);

    if (totalObjectsVolume + currentObjectVolume > boxVolume || currentObjectVolume === -1) {
        rl.close();
        return;
    }

    totalObjectsVolume += currentObjectVolume;
});

rl.on("close", () => {
    console.log(totalObjectsVolume);
});
